vm/src/stdlib/itertools.rs

```diff
@@ -736,6 +737,143 @@ impl PyItertoolsTee {
     }
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct PyIterToolsProduct {
+    pools: Vec<Vec<PyObjectRef>>,
+    idxs: RefCell<Vec<usize>>,
+    cur: RefCell<usize>,
```

Comment:
Is this separate `sizes` Vec necessary? It seems like it's just the length of the Vec at the same index in `pools`.

Reply:
Well, it can be implemented without separate `sizes` but i don't know what's the benefit of not separating. Is there special reason, @coolreader18 ?